Jenkinsfile

```diff
@@ -11,109 +11,192 @@ pipeline {
     environment {
         DOCS = 'distribution/docs'
         ITESTS = 'distribution/test/itests/test-itests-alliance'
+        POMFIX = 'libs/libs-pomfix,libs/libs-pomfix-run'
+        LARGE_MVN_OPTS = '-Xmx8192M -Xss128M -XX:+CMSClassUnloadingEnabled -XX:+UseConcMarkSweepGC '
+        LINUX_MVN_RANDOM = '-Djava.security.egd=file:/dev/./urandom'
     }
     stages {
         stage('Setup') {
             steps {
                 slackSend color: 'good', message: "STARTED: ${JOB_NAME} ${BUILD_NUMBER} ${BUILD_URL}"
             }
         }
-        stage('Parallel Build') {
+        // Use the pomfix tool to validate that bundle dependencies are properly declared
+        stage('Validate Poms') {
+            agent { label 'linux-small' }
+            steps {
+                retry(3) {
+                    checkout scm
+                }
+                withMaven(maven: 'M3', jdk: 'jdk8-latest', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings', mavenOpts: '${LINUX_MVN_RANDOM}') {
+                    sh 'mvn clean install -DskipStatic=true -DskipTests=true -pl $POMFIX'
+                }
+            }
+        }
+        // The incremental build will be triggered only for PRs. It will build the differences between the PR and the target branch
+        stage('Incremental Build') {
+            when {
+                allOf {
+                    expression { env.CHANGE_ID != null }
+                    expression { env.CHANGE_TARGET != null }
+                }
+            }
             // TODO CAL-296 refactor this stage from scripted syntax to declarative syntax to match the rest of the stages - https://issues.jenkins-ci.org/browse/JENKINS-41334
             steps {
                 parallel(
-                    linux: {
-                        node('linux-large') {
-                            retry(3) {
-                                checkout scm
+                        linux: {
+                            node('linux-large') {
+                                retry(3) {
+                                    checkout scm
+                                }
+                                timeout(time: 1, unit: 'HOURS') {
```

Comment:
@oconnormi will 1 hour be enough? I've seen several PR builds go for over 3 hours depending on how many modules were impacted in the dependency tree calculated by GIB. Maybe that's just because of the static analysis etc...?

Reply:
Alliance really should be faster than the DDF build. If this is increased, it shouldn't be by much and we should probably start making well documented decisions on why we are increasing the build time if an increase is needed in the future.

Reply:
I don't think I picked the 1 hour timeout. Pretty sure that was the value already being used in this pipeline. I can definitely increase it if we think it is necessary, but I think this build is relatively quick